compiler/testData/diagnostics/tests/smartCasts/inference/kt39010.fir.kt

```diff
@@ -0,0 +1,11 @@
+class A<E> {
+    fun foo(): E = TODO()
+}
+
+class B(var a: A<*>?) {
+    fun bar() {
+        if (a != null) {
+            a<!UNSAFE_CALL!>.<!>foo()
```

Comment:
This should have been SMARTCAST_IMPOSSIBLE. But the check in https://github.com/JetBrains/kotlin/blob/0946b4c5cb97fe0684fc7943b95533dc9c99bd4f/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/ResolutionStages.kt#L116 thinks the smartcast type `A<*>` is not a subtype of `A<E>`. I am not sure how to make the constraint system honor such a free variable `E`.

Reply:
Hi @demiurg906 @mglukhikh , any idea how I can mark `E` as a free variable?

Reply:
Actually, I have worked around the issue in #4382 by reporting all candidates from the unstable smartcast type, regardless of dispatch receiver type. So I no longer need to do any type check here.